plugins/scanners/fossid/src/main/kotlin/FossIdNamingProvider.kt

```diff
@@ -116,8 +116,8 @@ class FossIdNamingProvider(
     }
 
     /**
-     * Replaces non-standard characters in branch name and trims its length to one that will not exceed
-     * maximum length of FossID scan ID, when combined with the rest of variables
+     * Replace characters in [branch] not matching `[a-zA-Z0-9-_]` with underscores and trim its length so that the
```

Comment:
Instead of repeating the regex here, how about extracting it to a constant (which anyway would be good to avoid recompiling it on each call), and referring to that constant here instead?

Reply:
While at it, the regex could be simplified to `[^\w-]`.

Reply:
I have added a commit on top for that.